source/gameengine/Rasterizer/RAS_OpenGLRasterizer/RAS_OpenGLLight.cpp

```diff
@@ -200,6 +198,57 @@ MT_Matrix4x4 RAS_OpenGLLight::GetShadowMatrix()
 	return mat;
 }
 
+void RAS_OpenGLLight::GetShadowBox(MT_Vector3 *box)
+{
+	GPULamp *lamp;
+
+	if ((lamp = GetGPULamp())) {
+		KX_LightObject *light = (KX_LightObject *)m_light;
+		if (light->GetLightData()->m_type == RAS_ILightObject::LIGHT_SUN) {
+
+			box[0][0] = box[1][0] = box[2][0] = box[3][0] = -GPU_lamp_frustum_size(lamp);
+			box[4][0] = box[5][0] = box[6][0] = box[7][0] = +GPU_lamp_frustum_size(lamp);
+			box[0][1] = box[1][1] = box[4][1] = box[5][1] = -GPU_lamp_frustum_size(lamp);
+			box[2][1] = box[3][1] = box[6][1] = box[7][1] = +GPU_lamp_frustum_size(lamp);
+			box[0][2] = box[3][2] = box[4][2] = box[7][2] = -FLT_MAX;
+			box[1][2] = box[2][2] = box[5][2] = box[6][2] = -GPU_lamp_clipstart(lamp);
+
+			MT_Matrix3x3 orientation = light->NodeGetWorldOrientation();
```

Comment:
You could share the matrix/transform creation and apply to corner for sun light and spot light.